core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -4195,6 +4195,23 @@ public void doRestart(StaplerRequest req, StaplerResponse rsp) throws IOExceptio
             rsp.sendRedirect2(".");
         }
     }
+    
+    /**
+     * REST client version of {@link #doRestart(StaplerRequest, StaplerResponse)}
+     * Perform a restart of Jenkins, if we can.
+     *
+     * This first replaces "app" to {@link HudsonIsRestarting}
+     *
+     * @since TODO
+     */
+    @RequirePOST
+    public void doRestartSimple(StaplerRequest req, StaplerResponse rsp) throws RestartNotSupportedException {
```

Comment:
I've lost the other PR where I suggested you can identify a REST client by use of token based authentication.
I think that would be cleaner and this seems to almost be duplicating endoints which will potentially confuse.

in effect if the user POSTS to `/restart` then there is no difference.  If they sent  `GET` they would get a page back rather than a method not allowed / not found?

if you just want this behaviour difference why not have one end point that calls 2 different functions based on the caller being authenticated by a token?

Reply:
https://github.com/jenkinsci/jenkins/pull/4022#pullrequestreview-237335031?

Reply:
Especially my reply: https://github.com/jenkinsci/jenkins/pull/4022#discussion_r284121397 saying the token is not mandatory when you are using no security realm.